### PR TITLE
 Added "code:registry_api_request_count:rate:sum" recording rule 

### PR DESCRIPTION
**Pull request by abyrne55**
OpenShift SRE needs to be able to track the rates of API requests to the image registry service for Service Level Agreement compliance. The existing "code:apiserver_request_count:rate:sum" rule doesn't differentiate between the service each API call is going to, so it isn't specific enough for this purpose.

To remedy this, I added another recording rule that selects only the API call counters that correspond to the registry service, and then performs a rate-and-sum similar to "code:apiserver_request_count:rate:sum"

### CODE REVIEW

**Comment by abyrne55** [comment]
Seems like CI got rate-limited. Requesting a /retest

File: assets/prometheus-k8s/rules.yaml [hunk 1]
```diff
@@ -425,6 +425,9 @@ spec:
     rules:
     - expr: sum(rate(apiserver_request_count{job="apiserver"}[10m])) BY (code)
       record: code:apiserver_request_count:rate:sum
+    - expr: sum(rate(apiserver_request_count{job="apiserver",resource=~"image.*",verb!="WATCH"}[10m]))
```
Comment (brancz) [inline assets/prometheus-k8s/rules.yaml]: If it is intentional to only record the image resource here then that should be reflected in the name of the recording rule.

Comment (brancz) [reply assets/prometheus-k8s/rules.yaml]: :facepalm: you already did that. Out of curiosity, why specifically this API?

Comment (abyrne55) [reply assets/prometheus-k8s/rules.yaml]: Managed OSDv4 instances have an SLO regarding "the proportion of successful responses to API requests to the registry as measured from Prometheus"

Comment (smarterclayton) [reply assets/prometheus-k8s/rules.yaml]: This query doesn't select only requests made by the registry.  Lots of other components in openshift make many calls to the API.  

What are you trying to measure here?  What you're actually measuring is "how many API requests are made to things called image" which includes many things the registry has no impact on.

Comment (abyrne55) [reply assets/prometheus-k8s/rules.yaml]: AFAIK, the goal here is to measure "the proportion of successful responses to API requests to the registry as measured from Prometheus" 
@jewzaam might be able to explain the objective with this SLI better than I can though

**Pull request merged**